common/src/main/java/io/novafoundation/nova/common/mixin/condition/ConditionMixin.kt

```diff
@@ -0,0 +1,23 @@
+package io.novafoundation.nova.common.mixin.condition
+
+import androidx.annotation.StringRes
+import io.novafoundation.nova.common.presentation.DescriptiveButtonState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+
+interface ConditionMixin {
+
+    val buttonState: Flow<DescriptiveButtonState>
+
+    fun checkCondition(index: Int, isChecked: Boolean)
+}
+
+interface ConditionMixinFactory {
+
+    fun createConditionMixin(
+        coroutineScope: CoroutineScope,
+        conditionsCount: Int,
+        @StringRes enabledButtonText: Int,
+        @StringRes disabledButtonText: Int,
```

Comment:
Do we really need to make butten texts a part of mixin implementation? Imho we can keep interface minimal and just have `allConditionsSatisfied: Flow<Boolean>` there. Then, we may have an extension `buttonState(enabledText, disabledText): Flow< DescriptiveButtonState >`.This way we
1. Better conform to single responsibility
2. Dont force clients of ConditionMixin to always setup a button